prov/verbs/src/verbs_cq.c

```diff
@@ -441,20 +441,23 @@ static int fi_ibv_cq_close(fid_t fid)
 
 	cq = container_of(fid, struct fi_ibv_cq, cq_fid.fid);
 
+	util_buf_pool_destroy(cq->epe_pool);
+	util_buf_pool_destroy(cq->wce_pool);
+
 	if (atomic_get(&cq->nevents))
 		ibv_ack_cq_events(cq->cq, atomic_get(&cq->nevents));
 
 	fastlock_acquire(&cq->lock);
 	while (!slist_empty(&cq->wcq)) {
 		entry = slist_remove_head(&cq->wcq);
 		wce = container_of(entry, struct fi_ibv_wce, entry);
-		util_buf_release(cq->domain->fab->wce_pool, wce);
+		util_buf_release(cq->wce_pool, wce);
```

Comment:
You're releasing buffers to a pool that you just destroyed above.  The destroy calls need to move down.